### PR TITLE
Fix Flower Gift

### PR DESCRIPTION
**Pull request by Pikachuun**
[10:32:54] &xfix: (and fix Flower Gift... I should have done this long time)

Basically just doing this for xfix since he hasn't done it yet.

### CODE REVIEW

**Comment by kotarou3** [comment]
Duplicate of #2067 ?

**Comment by Pikachuun** [comment]
Yeah, I didn't notice that while creating this, my mistake.
I've tested this on the sim and it seems to work in the case that @Marty-D has described in his comment after making a quick update [Cherrim transforming into Flower Gift X], so there's that at least. Feel free to close if need be, xfix was first, after all.

**Comment by Rhythms88** [comment]
[14:50] @Marty: if you read my comment, you'll see that no, changing the species check like that isn't actually correct
[14:50] +Pikachuun: yeah I was just reading that and was about to ask you if my pull req would be accepted if I made it baseTemplate over template
[14:52] +Pikachuun: I'll do further testing.
[14:52] @Marty: xfix's comes with a test, so probably not

**Comment by Pikachuun** [comment]
I'm working on adding a test file.

**Comment by ascriptmaster** [comment]
I don't like using proportions like that to check stat changes because of how the damage calculations work. `BattlePokemon.getStat('atk')` is a much more reliable method and allows you to check Special Defense as well.

**Comment by Pikachuun** [comment]
@ascriptmaster, after running my test file and getting a failed attempt, I decided to test out that method in battle; `BattlePokemon.getStat('atk')

**Comment by ascriptmaster** [comment]
@Pikachuun, I see the problem now. It's a bug in `getStat`. #2079 will fix it when it goes through.

**Comment by ascriptmaster** [comment]
Ok, if you rebase and try with `getStat` again it should work.

**Comment by ascriptmaster** [comment]
Pikachuun, I said rebase, not merge....

**Comment by Pikachuun** [comment]
I couldn't seem to rebase it without something breaking (either that or I'm just paranoid), my mistake. I'll attempt to fix it.

**Comment by ascriptmaster** [comment]
Things are breaking anyways. You have trailing whitespace

**Comment by Pikachuun** [comment]
Working on it, the trailing whitespace issue is fixed but I need to rebase successfully.

**Comment by Zarel** [comment]
Travis seems to disagree about your trailing whitespace.

**Comment by Pikachuun** [comment]
I could've sworn I fixed it ._.

It's fixed now though

File: test/simulator/abilities/flowergift.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		]);
+		battle.commitDecisions();
+		var baseAtk = battle.p1.active[1].getStat('atk');
+		var baseSpd = battle.p1.active[1].getStat('spd');
+
+		// Set the weather to sun and re-check
+		battle.setWeather('sunnyday');
+		assert.strictEqual(battle.p1.active[1].getStat('atk'), battle.modify(baseAtk, 1.5));
+		assert.strictEqual(battle.p1.active[1].getStat('spd'), battle.modify(baseSpd, 1.5));
+	});
```
Comment (ascriptmaster) [inline test/simulator/abilities/flowergift.js]: We tend to put an extra line break between test cases for readability

File: test/simulator/abilities/flowergift.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			{species: "Cherrim", ability: 'flowergift', moves: ['healbell']},
+			{species: "Snorlax", ability: 'immunity', moves: ['healbell']}
+		]);
+		battle.join('p2', 'Guest 2', 1, [
+			{species: "Blissey", ability: 'serenegrace', moves: ['healbell']},
+			{species: "Blissey", ability: 'serenegrace', moves: ['healbell']}
+		]);
+		battle.commitDecisions();
+		var baseAtk = battle.p1.active[1].getStat('atk');
+		var baseSpd = battle.p1.active[1].getStat('spd');
```
Comment (ascriptmaster) [inline test/simulator/abilities/flowergift.js]: Maybe check Cherrim's stats as well

**Comment by Pikachuun** [comment]
Did the things you requested.

File: test/simulator/abilities/flowergift.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			{species: "Blissey", ability: 'serenegrace', moves: ['healbell']},
+			{species: "Blissey", ability: 'serenegrace', moves: ['healbell']}
+		]);
+		battle.commitDecisions();
+		var cherAtk = battle.p1.active[0].getStat('atk');
+		var cherSpd = battle.p1.active[0].getStat('spd');
+		var baseAtk = battle.p1.active[1].getStat('atk');
+		var baseSpd = battle.p1.active[1].getStat('spd');
+
+
```
Comment (ascriptmaster) [inline test/simulator/abilities/flowergift.js]: No! The extra line doesn't go here! It goes in between each of your `it` test cases! Now here has one too many!

Look at the other test files to get a sense of what I'm talking about.

**Comment by Pikachuun** [comment]
well I'm dumb

**Comment by ascriptmaster** [comment]
Run `npm test` (or at least `gulp`) *before* you push a new version please

**Comment by Pikachuun** [comment]
I've run npm test (twice for safety) this time and this works according to it.

I apologize for updating so many times to fix things that I really should have fixed beforehand (such as whitespace, etc.).

**Pull request merged**